string-manipulation-3/src/is-anagram.js

```diff
@@ -1,52 +1,38 @@
 /* exported isAnagram */
 /*
-create a storage container
-i want to figure out how many times a letter appears in a word
-i should go through the first word, and if that letter has not appeared before i would likes to send it to the storage container and say that it appeared once
-if i continue traversing the word and run into it again
-then i update the storage container to say that it appeated twice
-do this until you run out of letters to check
-then make a second storage container and repeat the process for the second word
-
-now to compare the storage containers i will compare the first letter's appearance count to the same letter's appearance count in the second container
-create a ledger that keeps track of truth/false
-if the counts are equal && both words have the same amount of letters then set the leger variable equal to true
-or else set it to false
-after a letter is evaluated check the value of the ledger
-if its true do nothing,
-if it's false return false
-if youre able to traverse the entire word successfully then return true
+remove spaces from the strings
+check if the lengths of both strings are equal
+if not then auto return false
+else continue
+turn both strings into arrays
+loop through array 1
+if array at index is found in array 2
+remove it from array 2
+else
+return false
+if array 2 is empty after the loop has ended
+return true
 */
 
 function isAnagram(firstString, secondString) {
-  var container = {};
-  var container2 = {};
   var newSecondString = secondString.replace(/ /g, '');
   var newFirstString = firstString.replace(/ /g, '');
+  var arrFirst = newFirstString.split('');
+  var arrSecond = newSecondString.split('');
```

Comment:
If you sort these, you can just re-join them and compare the results. It will eliminate the rest of the algorithm.

Reply:
cool thankS, had to dig into the MDN's for sort and join. hadn't used them till now.. TY